7a_temp_coop_munge/src/data_parsers/parse_mndnr_files.R

```diff
@@ -44,3 +44,108 @@ parse_mndnr_files <- function(inind, outind) {
+                  Result, Upper_Depth) %>%
+    dplyr::rename(DateTime = Date,
+                  time = Time,
+                  depth = Upper_Depth,
+                  temp = Result,
+                  DOW = DOWLKNUM,
+                  site = Lake_Name
+    ) %>%
+    mutate(DateTime = as.Date(strptime(DateTime, "%m/%d/%Y")),
+           time = substr(as.character(time), 1, 5))
+
+  return(out)
+}
+
+
+# parser and helper functions for Long29016100.zip -----------
+parse_Long_29016100 <- function(inind, outind) {
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # Unzip the file with all of Bull Shoals, then cleanup unzipped files
+  # which aren't needed externally before leaving the function.
+  unzip_dir <- tempdir()
+  files_from_zip <- unzip(infile, exdir = unzip_dir)
+  on.exit(unlink(unzip_dir, recursive = TRUE))
+
+
+  clean <- files_from_zip %>%
+    purrr::map_dfr(~ clean_Long29016100(file_path = .x,
+                                        skip = 2, date_cell = 'E1',
+                                        omit_sheets = c('Summary', 'Depths',
+                                                        'Layers', '8-12-16 (Sta 1 So.)',
+                                                        '8-12-16 (Sta 3 No.)')))
+
+  saveRDS(object = clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+}
+
+#' @param file_path chr, full file path for file
+#' @param skip int, number of rows to skip
+#' @param date_cell chr, Excel-style cell ID for the sample date
+#' @param omit_sheets chr, vector of sheets that should be omitted if encountered
+#'
+clean_Long_29016100 <- function(file_path, skip = 2,
+                               date_cell = 'E1', omit_sheets = NULL) {
+
+  # read sheets and remove omitted sheets
+  sheets <- readxl::excel_sheets(file_path) %>%
+    .[!(. %in% omit_sheets)]
+
+  clean <- sheets %>%
+    map_dfr(~ build_sheet_data(file_path = file_path, sheet = .x,
+                               date_cell = date_cell, skip = skip))
+}
+
+#' @param file_path chr, full file path for file
+#' @param sheet chr, name of the sheet that should be read in
+#' @param skip int, number of rows to skip
+#' @param date_cell chr, Excel-style cell ID for the sample date
+#'
+build_sheet_data <- function(file_path, sheet, date_cell, skip) {
+  # extract sample date
+  sample_date <- readxl::read_excel(path = file_path,
+                                    sheet = sheet,
+                                    range = date_cell,
+                                    col_names = F) %>% dplyr::pull()
+
+  # read sheet data, tack on date and DOW
+  dat <- readxl::read_excel(path = file_path, sheet = sheet, skip = skip) %>%
+    dplyr::select(matches('depth m|oC'))
+
+  names(dat) <- c('depth', 'temp') # doing it this way because the name of the depth column varies between datasets, but is always in the first position
```

Comment:
I think you can actually just rename based on a position if you put a number instead!

```r
iris %>% rename(new_name = 3)
    Sepal.Length Sepal.Width new_name Petal.Width    Species
1            5.1         3.5      1.4         0.2     setosa
2            4.9         3.0      1.4         0.2     setosa
3            4.7         3.2      1.3         0.2     setosa
4            4.6         3.1      1.5         0.2     setosa
```